### PR TITLE
fix tag indexing issue of a keep kept again after unkept

### PR DESCRIPTION
**Pull request by ymatsuda**
@eishay @andrewconner 

We will update the collection index when internKeep has resurrected an inactive keep.

### CODE REVIEW

**Comment by andrewconner** [comment]
Looks good. :)

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala [hunk 1]
```diff
@@ -173,17 +175,24 @@ class KeepInterner @Inject() (
...
-        (false, persistedKeep)
+        (false, wasInactiveKeep, persistedKeep)
       case None =>
         val urlObj = urlRepo.get(url, uri.id.get).getOrElse(urlRepo.save(URLFactory(url = url, normalizedUriId = uri.id.get)))
-        (true, keepRepo.save(KeepFactory(url, uri, userId, title orElse uri.title, urlObj, source, isPrivate, installationId)))
+        (true, false, keepRepo.save(KeepFactory(url, uri, userId, title orElse uri.title, urlObj, source, isPrivate, installationId)))
     }
+    if (wasInactiveKeep) {
+      // A inactive keep may have had tags already. Index them if any.
+      keepToCollectionRepo.getCollectionsForKeep(internedKeep.id.get) foreach { cid => collectionRepo.collectionChanged(cid) }
```
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala:192]: can it be in a future?

Comment (ymatsuda) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala:192]: I want this to be in the same transaction. Note that this line is not invoking indexing on search instances.

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala:192]: i see, lgtm

**Pull request merged**